Packs/ServiceNow/Integrations/ServiceNowEventCollector/ServiceNowEventCollector.py

```diff
@@ -216,37 +315,46 @@ def main() -> None:  # pragma: no cover
             url=server_url,
             verify=verify_certificate,
             proxy=proxy,
-            fetch_limit=max_fetch,
             api_server_url=api_server_url,
+            fetch_limit_audit=max_fetch_audit,
+            fetch_limit_syslog=max_fetch_syslog
         )
 
         if client.sn_client.use_oauth and not get_integration_context().get("refresh_token", None):
             client.sn_client.login(username=user_name, password=password)
 
         if command == "test-module":
-            return_results(module_of_testing(client))
+            return_results(module_of_testing(client, log_types))
 
         elif command == "service-now-get-audit-logs":
-            audit_logs, results = get_audit_logs_command(client=client, args=args)
+            last_run = demisto.getLastRun()
+            audit_logs, results = get_events_command(client=client, args=args, log_type=AUDIT, last_run=last_run)
             return_results(results)
 
             if argToBoolean(args.get("should_push_events", "true")):
                 send_events_to_xsiam(audit_logs, vendor=VENDOR, product=PRODUCT)
 
+        elif command == "service-now-get-syslog-transactions":
+            last_run = demisto.getLastRun()
+            syslog_logs, results = get_events_command(client=client, args=args, log_type=SYSLOG_TRANSACTIONS, last_run=last_run)
+            return_results(results)
+
+            if argToBoolean(args.get("should_push_events", "true")):
+                send_events_to_xsiam(syslog_logs, vendor=VENDOR, product=PRODUCT)
+
         elif command == "fetch-events":
             last_run = demisto.getLastRun()
             demisto.debug(f"Starting new fetch with last_run as {last_run}")
-            audit_logs, new_last_run = fetch_events_command(client=client, last_run=last_run)
+            events, next_run = fetch_events_command(client=client, last_run=last_run, log_types=log_types)
 
             demisto.debug("Done fetching events, sending to XSIAM.")
 
-            if audit_logs:
-                add_time_field(audit_logs)
-                send_events_to_xsiam(audit_logs, vendor=VENDOR, product=PRODUCT)
-                if new_last_run:
+            if events:
```

Comment:
take this code outside all the if's since its identical to all commands. for the send_events_to_xsiam part.
```suggestion
            if events or argToBoolean(args.get("should_push_events", True)):
```